aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -589,9 +588,6 @@ struct ReduceOp {
       for (index_t i = 0; i < vt0; i++) {
         const auto offset = calc(idx + i * stride) / output_vec_size;
         values[i] = memory::load_vector<output_vec_size>(data_, offset);
-      }
-      #pragma unroll
-      for (index_t i = 0; i < vt0; i++) {
```

Comment:
Did you check if this change affected performance? 

Background: For pointwise ops, the two different ways of writing code leads to very different perf:
```CUDA
#pragma unroll
for (int i = 0; i < 4; i++) {
  load
  compute
  store
}
```
vs
```CUDA
#pragma unroll
for (int i = 0; i < 4; i++) {
  load
}
#pragma unroll
for (int i = 0; i < 4; i++) {
  compute
}
#pragma unroll
for (int i = 0; i < 4; i++) {
  store
}
```
I don't think we have the same issue here, but worth checking.

Reply:
That's actually very fair. Sorry for the sloppy edit. I just rolled it back to be on the safe side.